app/models/micropost.rb

```diff
@@ -0,0 +1,21 @@
+class Micropost < ApplicationRecord
+  belongs_to :user
+
+  validates :user_id, presence: true
+  validates :content, presence: true,
+    length: {maximum: Settings.max_micropost_length}
+  validate  :picture_size
+
+  mount_uploader :picture, PictureUploader
+
+  default_scope ->{order created_at: :desc}
```

Comment:
tìm hiểu và so sánh `scope`, `default_scope`

Reply:
### `scope`
`scope` là các class method giúp thao tác lấy hay chọn đối tượng của model. Scope có thể được định nghĩa mới với tên riêng để sử dụng lại nhiều lần.
```ruby
class User < ApplicationRecord
  scope :activated, -> {where activated: true}
end

# Sử dụng
User.activated
```
Ngoài ra, scope có thể được dùng kết hợp với các scope khác. Các scope sẽ được kết hợp với nhau bằng phép `and`.
```ruby
class User < ApplicationRecord
  scope :active, -> { where state: "active" }
  scope :finished, -> { where state: "finished" }
end

# Sử dụng
User.active.finished
# => SELECT "user".* FROM "users" WHERE "users"."state" = "active" AND "users"."state" = "finished"
```

### `default_scope`
`default_scope` là một macro giúp đặt scope mặc định cho mọi thao tác lên model.
```ruby
class Article < ActiveRecord::Base
  default_scope {where published: true}
end

Article.all # => SELECT * FROM articles WHERE published = true
```
Nếu có nhiều `default_scope` trên model thì chúng sẽ "nhập" (merge) lại với nhau.
```ruby
class Article < ActiveRecord::Base
  default_scope {where published: true}
  default_scope {where rating: "G"}
end

Article.all # => SELECT * FROM articles WHERE published = true AND rating = "G"
```